src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+      <div
+        :id="multiselectSelectedItemsStagingId"
+        :key="stagingKey"
+        :style="widthStyle"
+        tabindex="-1"
+        aria-hidden="true"
+        class="k-multiselect-selections staging"
+      >
+        <KBadge
+          v-for="item in visibleSelectedItemsStaging"
+          :key="`${item ? item.key : ''}-badge`"
+          shape="rectangular"
+          dismissable
+          hidden
+          class="mr-1 mt-2"
+        >
+          {{ item ? item.label : '' }}
+        </KBadge>
+        <!-- Always render this badge even if it's hidden to ensure there will be
+             enough space to show it
+        -->
+        <KBadge
+          shape="rectangular"
+          hidden
+          class="mt-2 hidden-selection-count"
+        >
+          +{{ invisibleSelectedItemsStaging.length }}
+        </KBadge>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, Ref, computed, watch, PropType, nextTick } from 'vue'
+import { v1 as uuidv1 } from 'uuid'
+import useUtilities from '@/composables/useUtilities'
+import KBadge from '@/components/KBadge/KBadge.vue'
+import KButton from '@/components/KButton/KButton.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KInput from '@/components/KInput/KInput.vue'
+import KLabel from '@/components/KLabel/KLabel.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import KToggle from '@/components/KToggle'
+import KMultiselectItem from '@/components/KMultiselect/KMultiselectItem.vue'
+
+const { getSizeFromString, cloneDeep } = useUtilities()
+const SELECTED_ITEMS_SINGLE_LINE_HEIGHT = 34
+
+const defaultKPopAttributes = {
+  hideCaret: true,
+  placement: 'bottomStart',
+  popoverTimeout: 0,
+  popoverClasses: 'k-multiselect-popover mt-0',
+}
+
+export interface MultiselectItem {
+  label: string
+  value: string
+  key?: string
+  selected?: boolean
+}
+
+export interface MultiselectFilterFnParams {
+  items: MultiselectItem[]
+  query: string
+}
+
+export default defineComponent({
+  name: 'KMultiselect',
+  components: {
+    KBadge,
+    KButton,
+    KIcon,
+    KInput,
+    KLabel,
+    KPop,
+    KMultiselectItem,
+    KToggle,
+  },
+  inheritAttrs: false,
+  props: {
+    modelValue: {
+      type: Array as PropType<string[]>,
+      default: () => [],
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    labelAttributes: {
+      type: Object,
+      default: () => ({}),
+    },
+    placeholder: {
+      type: String,
+      default: '',
+    },
+    kpopAttributes: {
+      type: Object,
+      default: () => ({
+        popoverClasses: '',
+      }),
+    },
+    dropdownMaxHeight: {
+      type: String,
+      default: '300',
+    },
+    /**
+     * The width of the multiselect and popover's min-width
+     */
+    width: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Number of rows of selections to show when focused
+     */
+    selectionRowCount: {
```

Comment:
nit: Since we call items `selected` or not, can we change this to `selectedRowCount` ?